pkg/cmd/cli/cmd/helper.go

```diff
@@ -80,3 +80,22 @@ func selectString(s, spec string) bool {
 	}
 	return match
 }
+
+// parseValue parses the value and returns namespace, resource and the
+// value (resource name) itself. The valid syntax is:
+// oc process mytemplate - implicit namespace (current), implicit resource (template)
+// oc process template/mytemplate - implicit namespace (current), explicit resource
+// oc process ns/template/mytemplate - explicit namespace, explicit resource
+// oc process ns//mytemplate - explicit namespace, implicit resource (template)
+func parseValue(v string) (ns, resource, name string) {
```

Comment:
this may be the most generic name I have ever seen :)
can we be more specific? parseNamespaceResourceName
also, if we want this to handle the implicit cases, should we pass in defaultNamespace and defaultResource?